src/content/docs/dropins/all/extend-or-build.mdx

```diff
@@ -0,0 +1,148 @@
+---
+title: Build a new drop-in or extend an existing one?
+description: A decision framework to determine when to create a new drop-in component versus extending or composing existing drop-ins, including granularity guidelines, examples, and escalation paths.
+sidebar:
+  label: Build vs. extend a drop-in
+---
+
+import Vocabulary from '@components/Vocabulary.astro';
+import Diagram from '@components/Diagram.astro';
+import Aside from '@components/Aside.astro';
+import Callouts from '@components/Callouts.astro';
+import { Steps } from '@astrojs/starlight/components';
+import Tasks from '@components/Tasks.astro';
+import Task from '@components/Task.astro';
+import { Tabs, TabItem } from '@astrojs/starlight/components';
+
+Developers often ask: “Should I build a new drop-in, or just add my functionality by extending the existing ones?” This guide provides a simple decision process, focusing on when NOT to extend a drop-in using the provided methods: slots, configuration, style overrides, and public events.
+
+## Vocabulary
+
+<Vocabulary>
+
+### Extending a drop-in
+
+Adding functionality to a drop-in by adding slots, configuration, style overrides, and public events.
+
+### Building a new drop-in
+
+Building a new drop-in is the process of creating a new drop-in from scratch.
+
+### Slots
+
+Declarative anchor where arbitrary UI content can be injected.
+
+### Configuration
+
+Configuration is the process of setting the properties for a drop-in container.
+
+### Style overrides
+
+Style overrides are the process of overriding the styles of a drop-in container.
+
+### Public events
+
+Public events are the process of firing an event from a drop-in.
+
+</Vocabulary>
+
+## Decision flow
+
+Need a simple mermaid flowchart. This one is just a placeholder.
+
+<Diagram
+  type="mermaid"
+  caption="Decision Flow"
+  code={`
+flowchart TD
+    A[Start] --> B{Is this a one-off customization?}
+    B -- Yes --> C[Build local component or extend via slot]
+    B -- No --> D{Does an existing drop-in cover ≥60% of the domain?}
+  `}
+/>
+
+## When NOT to build a new drop-in
+
+- Adding a provider-specific script (e.g., reviews widget, chat bubble, analytics pixel).
+- Injecting a promo banner into Cart or Checkout header.
+- Styling changes or minor UI variations.
```

Comment:
UI variations don't necessarily have to be minor; it depends on what's considered "minor." This implies that if I have major changes, I might need to build one, which is not the case. Some major customization can be done by restructuring the layout of the drop-in, swapping containers with custom frontend components, slots, etc.